### ISSUE TITLE
Session handover 2026-04-29: ztensor#103 verified, plan-sync done, one branch needs review

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#839]
## Purpose

Handover from a Claude Code session on a different machine. Picks up here.

## Current main-branch state (zerfoo/zerfoo)

- HEAD: `7539627a` (or newer) -- includes T124.2.3 activation unification + softmax fix.
- `docs/plan.md` is monolithic (4,005 lines). A split-layout migration was attempted earlier and reverted; do not redo without explicit user direction.

## What shipped this session

- **zerfoo/ztensor#104** -- bulk F32 weight upload path. Merged. Released as `v1.8.0`.
- **feza-ai/wolf#178** -- bumped ztensor v1.6.0 -> v1.8.0. Merged.
- Image `localhost:5000/wolf-train-crossasset:44fb4790` built on DGX via Spark buildah pod.
- **Verified live on GB10**: `ztensor103-verify-r3` ran `-folds=2 -epochs=2`, Fold 0 acc=0.6436 in 424.5s, 18m32s wall-clock, exit 0. Pre-upload no longer wedges.

## Closed issues

| Issue | Reason |
|---|---|
| zerfoo/ztensor#102 | not planned -- regression was in build env, not v1.6.0..v1.7.0 source |
| zerfoo/ztensor#103 | completed -- bulk-upload fix shipped + verified live |
| feza-ai/spark#32 | completed -- v1.13.0 fixed pending-with-no-events |
| feza-ai/spark#37 | completed -- reconciler leak handled |

## Open: pre-existing branch `chore/t124.2.4-tests-lint`

Local + `origin/chore/t124.2.4-tests-lint` carry 2 commits NOT on main:

```
6393a20b fix(timeseries): seed iTransformer init for deterministic gradient checks
a174bdb9 chore(plan): close out T124.2.4 tests+lint after activation unification
```

Status: **needs human review.** This was the branch the session started on; the iTransformer determinism fix is real test code, the plan close-out is documentation. /tidy preserved them rather than auto-PR-merging. Resume options:

1. Open a PR from this branch and review/merge if the changes are still wanted.
2. Confirm they are superseded and delete the branch.

## GitHub Project state (zerfoo/zerfoo Project #2)

- All 303 tracked tasks have `Component` + `Wave` + `Epic` + `Kind` fields populated.
- 278 tasks have a GitHub Milestone assigned (built-in field, not a custom select).
- 19 repo milestones exist (M1..M8 + M-LAYOUT-1, M-COMP-3/4, M-E86, M-E89, M-E90, M-E92, M-E125, M-E126, plus M1.5/M1.6).
- Manifest: `.claude/scratch/sync-manifest.json` is the single source.

## Skill changes (~/.claude/skills/plan/SKILL.md)

User extended the /plan skill with:

- Component convention: `**Component:** <name>` per epic file.
- Milestone convention: Timeline table rows of form `| M\d+ | label | T1.1, T1.2, ... | exit criteria |` (parser harvests task IDs from non-first cells).
- Bumped sync manifest schema with `milestone` + `component` field IDs and option maps.
- Migration documented under `--migrate-split`.

If you re-run `/sync` from a fresh checkout, these conventions are now baseline.

## Repo housekeeping done by /tidy

- 24 stranded agent worktrees removed.
- 24 stale local branches deleted (E124 wave work, all merged or already-landed under different SHAs).
- Remote tracking refs pruned via `git fetch --prune`.
- Active references in `.claude/scratch/`: `sync-manifest.json`, `usecases-manifest.json`, `parsed-plan.json`, `sync-manifest.user-project-2.bak` (backup of an earlier project).

## To resume on the new machine

1. `cd ~/Code/zerfoo/zerfoo && git fetch --all --prune && git checkout main && git pull`
2. `gh repo set-default zerfoo/zerfoo`
3. `gh auth status` -- confirm token covers zerfoo + feza-ai orgs.
4. Decide on the `chore/t124.2.4-tests-lint` branch (PR or delete).
5. If continuing E124 followups: refer to docs/plan.md sections E124, E125, E126 for remaining open tasks.

## Environment notes

- DGX Spark (192.168.86.250) on v1.13.0; HTTP API healthy. SSH from this machine has been flaky -- prefer Spark API + buildah pods over `ssh + podman build`.
- `scripts/build-train-crossasset-image.sh` is the legacy SSH-driven build path. The new Spark-pod buildah path is at Wolf SHA `2451feab` and works reliably.

Closes nothing. Reference issue only -- close when handover is acknowledged.

**Comment by dndungu** [comment]
Closing completed/archived epic bookkeeping per ADR-093 tracker-hygiene pass; history preserved in the issue itself and docs/devlog.md.
